flink-ml-lib/src/main/java/org/apache/flink/ml/feature/bucketizer/Bucketizer.java

```diff
@@ -0,0 +1,179 @@
+        String[] inputCols = getInputCols();
+        String[] outputCols = getOutputCols();
+        Double[][] splitsArray = getSplitsArray();
+
+        Preconditions.checkArgument(inputCols.length == outputCols.length);
+        Preconditions.checkArgument(inputCols.length == splitsArray.length);
+        for (Double[] splits : splitsArray) {
+            Preconditions.checkArgument(
+                    splits.length >= 3,
+                    "Illegal value for "
+                            + BucketizerParams.SPLITS_ARRAY
+                            + ". See param "
+                            + BucketizerParams.SPLITS_ARRAY
+                            + " for details.");
+            for (int j = 1; j < splits.length; j++) {
+                Preconditions.checkArgument(
+                        splits[j] > splits[j - 1],
+                        "Illegal value for "
+                                + BucketizerParams.SPLITS_ARRAY
+                                + ". See param "
+                                + BucketizerParams.SPLITS_ARRAY
+                                + " for details.");
+            }
+        }
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        TypeInformation<?>[] outputTypes = new TypeInformation[outputCols.length];
+        Arrays.fill(outputTypes, BasicTypeInfo.INT_TYPE_INFO);
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(inputTypeInfo.getFieldTypes(), outputTypes),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getOutputCols()));
+
+        DataStream<Row> result =
+                tEnv.toDataStream(inputs[0])
+                        .flatMap(
+                                new FindBucketFunction(inputCols, splitsArray, getHandleInvalid()),
+                                outputTypeInfo);
+        return new Table[] {tEnv.fromDataStream(result)};
+    }
+
+    /** Finds the bucket index for each continuous feature of an input data point. */
+    private static class FindBucketFunction implements FlatMapFunction<Row, Row> {
+        private final String[] inputCols;
+        private final String handleInvalid;
+        private final Double[][] splitsArray;
+
+        public FindBucketFunction(
+                String[] inputCols, Double[][] splitsArray, String handleInvalid) {
+            this.inputCols = inputCols;
+            this.splitsArray = splitsArray;
+            this.handleInvalid = handleInvalid;
+        }
+
+        @Override
+        public void flatMap(Row value, Collector<Row> out) {
+            Row outputRow = new Row(inputCols.length);
+
+            for (int i = 0; i < inputCols.length; i++) {
+                double feature = ((Number) value.getField(inputCols[i])).doubleValue();
```

Comment:
If the field value is not a Number, the operator would throw exception, which means that `HasHandleInvalid` is always `ERROR` in this case.

Reply:
`Bucketizer` has assumed that the input is always a `Number`. I think the default thrown exception is enough for debugging.

The `ERROR` is for two cases:
- `NaN` 
- The input values that do not fit in the given split arrays.